packages/modeling/src/operations/booleans/martinez/segment_intersection.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Implementation of the Martinez 2D polygon clipping algorithm
+ * Copyright (c) 2018 Alexander Milevski
+ * https://github.com/w8r/martinez
+ */
+
+/**
+ * Finds the magnitude of the cross product of two vectors (if we pretend
+ * they're in three dimensions)
+ *
+ * @param {Object} a First vector
+ * @param {Object} b Second vector
+ * @private
+ * @returns {Number} The magnitude of the cross product
+ */
+const crossProduct = (a, b) => (a[0] * b[1]) - (a[1] * b[0])
```

Comment:
I may be wrong but think I saw this one elsewhere in jscad, even multiple times. Not sure if it is wise to hunt for it now or maybe do as a separate PR.

Reply:
`vec2.cross` returns a `vec3` whereas this function actually returns the magnitude of the cross product. I am leaving this function for now to avoid unnecessary array allocations.

Reply:
ok, nice